tests/publiccloud/check_registercloudguest.pm

```diff
@@ -26,18 +26,17 @@ my $path = is_sle('=15-SP2') ? '/usr/sbin/' : '';    # 15-SP2 is the oldest vers
 my $regcode_param = (is_byos()) ? "-r " . get_required_var('SCC_REGCODE') : '';
 
 sub run {
-    my ($self, $args) = @_;
-    my ($provider, $instance);
+    my ($self) = @_;
+    my ($instance);
     select_host_console();
 
     $run_count++;
 
     if (get_var('PUBLIC_CLOUD_QAM', 0)) {
-        $instance = $self->{my_instance} = $args->{my_instance};
-        $provider = $self->{provider} = $args->{my_provider};    # required for cleanup
+        $instance = $self->instance;
     } else {
-        $provider = $self->provider_factory();
-        $instance = $self->{my_instance} = $provider->create_instance(check_guestregister => is_openstack ? 0 : 1);
+        $self->provider($self->provider_factory);
+        $instance = $self->provider->create_instance(check_guestregister => is_openstack ? 0 : 1);
```

Comment:
can we just hide this into `basetest` ? so ANY test module would always have `$self-instance` and `$self->provider` ready before `run` function called ?

Reply:
I'm not sure what you mean here.
Not every test module does something like that, right?
basetest should not have specific publiccloud stuff.

Reply:
Or maybe I get what you mean. You don't mean os-autoinst basetest, but `publiccloud::basetest`. What you could do in `lib/main_publiccloud.pm` before you call the `loadtest(...)`:
```
       publiccloud::basetest->provider($self->provider_factory);
       publiccloud::basetest->provider->create_instance(check_guestregister => is_openstack ? 0 : 1);
```

Reply:
> Or maybe I get what you mean. You don't mean os-autoinst basetest, but `publiccloud::basetest`. What you could do in `lib/main_publiccloud.pm` before you call the `loadtest(...)`:


 yes